measurePower.py

```diff
@@ -62,21 +62,25 @@ def stat(self, cmd: str, duration: int):
         self.exec_thread.start()
         logger.info(f"Running {self.cmd}")
 
-    def stop(self):
+    def stop(self) -> None:
         logger.info(f"Stopping measurePower execution on {self.pod_name}")
-        r = self.exec_thread.join()
-        if r.returncode != 0:
-            logger.error(r)
-        self._output = self.generate_output(data=r.out)
+        self.exec_thread.join()
+        if self.exec_thread.result is not None:
+            r = self.exec_thread.result
+            if r.returncode != 0:
+                logger.error(r)
+            self._output = self.generate_output(data=r.out)
+        else:
+            logger.error("Thread did not return a result")
```

Comment:
`self._output` should be set to an empty `PluginOutput`.

You already handle the error (by logging a message). If you leave `self._output` unset, a crash will later happen, when trying to access the output. The error is handled, and there must be some usable output set, so that followup code can do something with that.

The function later is refactored, but it still applies then.

Reply:
probably, just call `self.generate_output(data=None)` and those functions need to return something usable. 

`generate_output()` could also return `None` (an `Optional[common.BaseOutput]`). But then every user has to handle None specially. Better to return some output with no data.

Reply:
That is a good point. I added this line to the `stop` refactor: [self._output = common.BaseOutput("", {})](https://github.com/wizhaoredhat/ocp-traffic-flow-tests/blob/main/task.py#L125)